tests/js/client/shell/shell-med-queue-blocked-cluster.js

```diff
@@ -0,0 +1,145 @@
+
+let jsunity = require('jsunity');
+let arangodb = require('@arangodb');
+let db = arangodb.db;
+let { getEndpointsByType,
+      debugCanUseFailAt,
+      debugClearFailAt,
+      debugSetFailAt,
+    } = require('@arangodb/test-helper');
+      
+function medQueueBlockedSuite() {
+  'use strict';
+  const cn = 'UnitTestsMedQueueBlocked';
+
+  return {
+
+    setUp: function () {
+      // We are using the direct connection to set the failure point here,
+      // since for this particular failure point (block scheduler medium
+      // queue) a reconnect is not possible. Therefore we cannot use
+      // debugFailAt.
+      arango.DELETE_RAW("/_admin/debug/failat");
+      db._drop(cn);
+    },
+
+    tearDown: function () {
+      // We are using the direct connection to set the failure point here,
+      // since for this particular failure point (block scheduler medium
+      // queue) a reconnect is not possible. Therefore we cannot use
+      // debugFailAt.
+      arango.DELETE_RAW("/_admin/debug/failat");
+      require("internal").wait(3);  // give the system time to finish the 
+                                    // creation of the collection, in case
+                                    // the test TestCreateColl fails.
+      db._drop(cn);
+    },
+    
+    testRunAql: function() {
+      let coll = db._create(cn, {numberOfShards:3});
+      let l = [];
+      for (let i = 0; i < 100; ++i) {
+        l.push({"Hello":i});
+      }
+      coll.insert(l);
+
+      // We now want to overload the medium prio scheduler queue, to this
+      // end, we post some perverse queries asynchronously. This should
+      // overwhelm any server without the fix. Note that we post the query
+      // with x-arango-frontend: true, such that we can start enough queries
+      // concurrently to fill the medium queue on the coordinator.
+      const f = require("@arangodb/aql/functions")
+      try {
+        f.register("USER::FUNC", function() {
+          const db = require("@arangodb").db;
+          require("internal").wait(1);
+          db._query("FOR d IN @@c RETURN d", {"@c": "UnitTestsMedQueueBlocked"});
+          return 12;
+        })
+        let start = new Date();
+        let jobs = []; 
+        for (let i = 0; i < 100; ++i) {
+          jobs.push(arango.POST_RAW("/_api/cursor",
+            {query:`LET s = USER::FUNC() FOR d IN ${cn} RETURN {d,s}`},
+            {"x-arango-async": "store", "x-arango-frontend": true}).headers["x-arango-async-id"]);
+        }
+        let count = 0;
+        for (let j of jobs) {
+          while (true) {
+            let res = arango.PUT_RAW(`/_api/job/${j}`, {});
+            if (res.code !== 204) {
+              break;
+            }
+            if (new Date() - start > 50000) {
+              throw "Lost patience, almost certainly the coordinator is deadlocked!";
+            }
+            require("internal").wait(0.1);
+          }
+          count += 1;
```

Comment:
It only occurred to me now that we are increasing the variable `count` here, but whatever is in this variable is irrelevant.
Is that intentional?

Reply:
Will remove it, this was used for some debugging messages which are gone.